main.go

```diff
@@ -78,15 +50,38 @@ func main() {
 			panic(err)
 		}
 
-		postUnmarsh.Title = fmt.Sprintf("%.25s...", postUnmarsh.Title)
 		postUnmarsh.HoursSincePosting = postUnmarsh.AddHoursSincePosting()
 		postUnmarsh.FormattedTime = postUnmarsh.RelativeTime()
 
-		fmt.Println(i, postUnmarsh.Score, postUnmarsh.Author, postUnmarsh.Title, postUnmarsh.FormattedTime, "ago")
-	}
+		num := strconv.Itoa(i)
+		o := fmt.Sprintln(num, postUnmarsh.Score, postUnmarsh.Author, postUnmarsh.Title, postUnmarsh.FormattedTime, "ago")
+
+		totalOutputLen := utf8.RuneCountInString(o)
+		titleLen := utf8.RuneCountInString(postUnmarsh.Title)
+		fmt.Println("titleLen:", titleLen)
+		fmt.Println("totalOutputLen:", totalOutputLen)
```

Comment:
This is IMHO a bit too complex, the only field that you should shrink is the title.

So if you know the width of all other fields (and you should), you can compute the max title width from them.

E.g.

```
const dateWidth = 3
const indexWidth = 3
const idWidth = 3

var termWidth = // ...
var titleWidth = termWidth - dateWidth - indexWidth - idWidth
```

Reply:
It's basically what you've done, but you can do that only once for the whole print, no need to do it for each line.

Reply:
That being said, the code approaches a solution so I'm going to approve it for now and you can improve it in a future PR.